x-pack/plugins/reporting/server/config/schema.ts

```diff
@@ -9,7 +9,17 @@ import { ByteSizeValue, schema, TypeOf } from '@kbn/config-schema';
 import moment from 'moment';
 
 const KibanaServerSchema = schema.object({
-  hostname: schema.maybe(schema.string({ hostname: true })),
+  hostname: schema.maybe(
+    schema.string({
+      hostname: true,
+      validate(value) {
+        if (/^0.0.0.0$/.test(value)) {
```

Comment:
There are plenty of ways to hack that still:
```
0.0.0.0
0.0.0
0.0
0
::
0000:0000:0000:0000:0000:0000:0000:0000
```

Those are all valid the same IP addresses.
We already have a dependency on [`ipaddr.js`](https://www.npmjs.com/package/ipaddr.js) library. I think the best option will be to reuse that to check that the address is non-zero.

```typescript
import { sum } from 'lodash';
import ipaddr from 'ipaddr.js';

if (!sum(ipaddr.parse(value).toByteArray())) {
  return `must be a valid hostname, not "${value}"`;
}

```

Reply:
:facepalm: 

Thanks!!